sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/FileClientBuilder.java

```diff
@@ -121,7 +123,7 @@ public FileAsyncClient buildAsyncClient() {
         }
 
         if (sasTokenCredential == null && sharedKeyCredential == null) {
-            throw new IllegalArgumentException("Credentials are required for authorization");
+            logger.logAndThrow(new IllegalStateException("Credentials are required for authorization"));
```

Comment:
IllegalStateException usually thrown when something wrong invoked in runtime. It is more like a RuntimeException